sdks/python/apache_beam/io/fileio.py

```diff
@@ -221,22 +235,33 @@ def close(self, file_handle):
     if file_handle is not None:
       file_handle.close()
 
+  @check_accessible(['file_path_prefix', 'file_name_suffix'])
   def initialize_write(self):
-    tmp_dir = self.file_path_prefix + self.file_name_suffix + time.strftime(
+    file_path_prefix = self.file_path_prefix.get()
+    file_name_suffix = self.file_name_suffix.get()
+    tmp_dir = file_path_prefix + file_name_suffix + time.strftime(
         '-temp-%Y-%m-%d_%H-%M-%S')
+    if self._file_system is None:
+      self._file_system = get_filesystem(self._pattern)
```

Comment:
What is `self._pattern`? Should this be `get_filesystem(file_path_prefix)`? (same for other locations in this file)